extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -2,114 +2,61 @@ package etcdsnapshot
 	rancherv1 "github.com/rancher/shepherd/clients/rancher/v1"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
 	"github.com/rancher/shepherd/extensions/clusters"
 	"github.com/rancher/shepherd/extensions/defaults"
-	"github.com/rancher/shepherd/extensions/defaults/stevetypes"
 	"github.com/sirupsen/logrus"
 	kwait "k8s.io/apimachinery/pkg/util/wait"
 )
 
 const (
 	ProvisioningSteveResouceType = "provisioning.cattle.io.cluster"
+	SnapshotSteveResourceType    = "rke.cattle.io.etcdsnapshot"
+	SnapshotClusterNameLabel     = "rke.cattle.io/cluster-name"
 	fleetNamespace               = "fleet-default"
 	localClusterName             = "local"
 	active                       = "active"
 	readyStatus                  = "Resource is ready"
+	oneSecond                    = 1000000000
 )
 
-// GetRKE1Snapshots is a helper function to get the existing snapshots for a downstream RKE1 cluster.
-func GetRKE1Snapshots(client *rancher.Client, clusterName string) ([]management.EtcdBackup, error) {
-	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
-	if err != nil {
-		return nil, err
-	}
-
-	snapshotSteveObjList, err := client.Management.EtcdBackup.ListAll(&types.ListOpts{
-		Filters: map[string]interface{}{
-			"clusterId": clusterID,
-		},
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	snapshots := []management.EtcdBackup{}
-
-	for _, snapshot := range snapshotSteveObjList.Data {
-		if strings.Contains(snapshot.Name, clusterID) {
-			snapshots = append(snapshots, snapshot)
-		}
-	}
-
-	sort.Slice(snapshots, func(i, j int) bool {
-		return snapshots[i].Created > snapshots[j].Created
-	})
-
-	return snapshots, nil
-}
-
-// GetRKE2K3SSnapshots is a helper function to get the existing snapshots for a downstream RKE2/K3S cluster.
-func GetRKE2K3SSnapshots(client *rancher.Client, clusterName string) ([]rancherv1.SteveAPIObject, error) {
-	localclusterID, err := clusters.GetClusterIDByName(client, localClusterName)
-	if err != nil {
-		return nil, err
-	}
-
-	steveclient, err := client.Steve.ProxyDownstream(localclusterID)
-	if err != nil {
-		return nil, err
-	}
-
-	snapshotSteveObjList, err := steveclient.SteveType(stevetypes.EtcdSnapshot).List(nil)
-	if err != nil {
-		return nil, err
-	}
-
-	snapshots := []rancherv1.SteveAPIObject{}
-
-	for _, snapshot := range snapshotSteveObjList.Data {
-		if strings.Contains(snapshot.ObjectMeta.Name, clusterName) {
-			snapshots = append(snapshots, snapshot)
-		}
-	}
-
-	sort.Slice(snapshots, func(i, j int) bool {
-		return snapshots[i].ObjectMeta.CreationTimestamp.Before(&snapshots[j].ObjectMeta.CreationTimestamp)
-	})
+// CreateRKE1Snapshot is a helper function to create a snapshot on an RKE1 cluster. Returns error if any.
+func CreateRKE1Snapshot(client *rancher.Client, clusterName string) ([]management.EtcdBackup, error) {
 
-	return snapshots, nil
-}
+	updateTimestamp := time.Now().UTC()
+	logrus.Info("UpdateTime: ", updateTimestamp.String())
```

Comment:
You can probably just leave this out, but love that all of this can be cut down.

Reply:
ack